File: server/src/main/java/b305/coffeebrew/server/entity/Review.java
package b305.coffeebrew.server.entity;

import b305.coffeebrew.server.config.utils.BaseAtTime;
import b305.coffeebrew.server.config.utils.BooleanToYNConverter;
import b305.coffeebrew.server.dto.review.ReviewPageDTO;
import lombok.*;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.io.Serializable;

@DynamicInsert
@DynamicUpdate
@Entity
@Table(name = "review")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review extends BaseAtTime implements Serializable {

    // 식별번호
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idx;

    // 회원 식별
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_idx")
    private Member memberIdx;

    @NotBlank
    private String itemType;

    @NotNull
    private Long itemIdx;

    private String content;

    @NotNull
    private int overall; // 총점
    
    @NotNull
    private int flavor; // 향미

    @NotNull
    private int acidity; // 산미

    @NotNull
    private int sweetness; // 단맛
    @NotNull
    private int bitterness; // 쓴맛

    @NotNull
    private int body; // 바디감


    @NotBlank
    private String coffeeing_note; // 커핑노트(향 상세)

    @NotNull
    private int like; // 좋아요 수치

    @Convert(converter = BooleanToYNConverter.class)
    private boolean expired;

    @Override
    public void prePersist() {
        super.prePersist();
    }

    public void update(ReviewPageDTO reviewPageDTO){
        this.content = reviewPageDTO.getContent();
        this.overall = reviewPageDTO.getOverall();
        this.flavor = reviewPageDTO.getFlavor();
        this.acidity = reviewPageDTO.getAcidity();
        this.sweetness = reviewPageDTO.getSweetness();
        this.bitterness = reviewPageDTO.getBitterness();
        this.body = reviewPageDTO.getBody();
        this.coffeeing_note = reviewPageDTO.getCoffeeing_note();
        this.like = reviewPageDTO.getLike();
    }
}

File: server-recom/app/db/models/Review.py
from sqlalchemy import Column
from sqlalchemy import Integer, String, DateTime

from ..database import Base


class Review(Base):
    def __getitem__(self, key):
        return getattr(self, key)

    def __setitem__(self, key, value):
        return setattr(self, key, value)

    def keys(self):
        return list(self.__dict__.keys())[1:]

    def values(self):
        return list(self.__dict__.values())[1:]

    __tablename__ = "review"

    idx = Column(Integer, primary_key=True, index=True)
    created_date = Column(DateTime())
    updated_date = Column(DateTime())
    acidity = Column(Integer)
    bitterness = Column(Integer)
    body = Column(Integer)
    coffeeing_note = Column(String(255))
    content = Column(String(255))
    expired = Column(String(255))
    flavor = Column(Integer)
    item_idx = Column(Integer)
    item_type = Column(String(255))
    like = Column(Integer)
    overall = Column(Integer)
    sweetness = Column(Integer)
    member_idx = Column(Integer)
